src/Header.php

```diff
@@ -1,18 +1,31 @@
 <?php
 
+declare(strict_types=1);
+
 namespace RicorocksDigitalAgency\Soap;
 
 use Illuminate\Contracts\Support\Arrayable;
+use SoapVar;
 
-class Header implements Arrayable
+final class Header implements Arrayable
 {
-    public $name;
-    public $namespace;
-    public $data;
-    public $actor;
-    public $mustUnderstand;
+    public string $name;
+
+    public string $namespace;
+
+    /**
+     * @var SoapVar|array<string, mixed>|null
+     */
+    public SoapVar|array|null $data;
+
+    public ?string $actor;
+
+    public bool $mustUnderstand;
 
-    public function __construct(?string $name = null, ?string $namespace = null, $data = null, bool $mustUnderstand = false, $actor = null)
+    /**
+     * @param SoapVar|array<string, mixed>|null $data
+     */
+    public function __construct(string $name = '', string $namespace = '', SoapVar|array $data = null, bool $mustUnderstand = false, string $actor = null)
```

Comment:
If we're bumping to PHP 8.x, shall we go for constructor promotion where viable?

Reply:
I think we just need to loosen the type on `actor`, [it can also be an `int`](https://www.php.net/manual/en/soap.constants.php). Either `string|int` or `mixed`.

Reply:
Updated